docs/prover/diagnosis/index.md

```diff
@@ -5,15 +5,238 @@ Diagnostic Tools
+
+
+```{note}
+The names of the TAC variables are often not helpful for understanding what they 
+mean. The valuation can nonetheless be useful for variables whose meaning is 
+clear from context (e.g. operations they occur in, or the source pointers).
+```
+
+### UNSAT TAC reports
+
+By default, the TAC report in the UNSAT case is the same as the plain TAC
+report. This changes when a `--coverage_info` option other than "NONE" is
+chosen.
+
+% TODO write about unsat core TAC reports, link to the other unsat core doc
+%  Jaroslav might be able to recycle some of his writings (?)
+
+### Timeout TAC reports
+
+In case of a Timeout result, the TAC report contains additional information that
+is meant to help with preventing the timeout in the future.
+
+```{figure} timeout-tac-report-overview.png
+:name: timeout tac report plain
+:align: center
+
+example TAC report for a run with a Timeout result
+```
+
+The figure above shows a TAC report that was generated from a Certora Prover run
+that timed out. Compared to the plain TAC report, additional information is
+available in three ways:
+ - Explanations and statistics are given in the box in the top right corner 
+   (double click to expand).
+ - CFG nodes have colors that indicate how the Prover run went and which parts 
+   are difficult.
+ - In the source code box there is difficulty information given by color 
+   highlights to some commands and by additional text giving difficulty 
+   statistics on difficult code blocks .
+
+
+
+#### Statistics- and explanation-box
+
+The box in the top right of the Timeout TAC reports contains explanations on the
+node colors as well as statistical information pertaining to how difficult the
+verification is to solve in general and what might be the particularly difficult
+parts.   
+
+```{figure} timeout-tac-report-explanation-box.png
+:name: timeout tac report explanation box
+:align: center
+
+Timeout TAC report with expanded statistics- and explanation-box
+```
+
+The above picture shows the expanded statistics- and explanation box. Currently,
+the contents of the box are the following:
+ - explanations for each color used in the CFG nodes
+ - general difficulty-related statistics pertaining to the whole program
+ - per-call breakdown of the "path count" statistic
+ - per-call breakdown of the "number of nonlinear operations" statistic
+
+
+#### Split- and heuristic difficulty-coloring in CFG and source code
+
+The fill-colors of the graph nodes in the timeout TAC report summarize the
+outcomes of the individual control flow splits for the current rule. See
+{ref}`control-flow-splitting` for background on the procedure.
+An additional violet gradient is used to indicate nodes that are likely to 
+contain code that is difficult for the Prover to handle.
+
+Nodes that have a green fill color are contained exclusively in splits that have
+been successfully proven correct by the current run of the Prover. This means
```

Comment:
`prover correct` or `proven UNSAT`? You use the UNSAT terminology above to cover also `satisfy` checks.

Reply:
yeah, UNSAT ✔️